gnovm/pkg/gnolang/nodes.go

```diff
@@ -1132,8 +1132,12 @@ func ReadMemPackage(dir string, pkgPath string) *std.MemPackage {
 				Body: string(bz),
 			})
 	}
-	validatePkgName(string(pkgName))
+
 	memPkg.Name = string(pkgName)
+	if err := memPkg.Validate(); err != nil {
+		panic(fmt.Sprintf("unable to validate mempkg package: %q", err))
+	}
```

Comment:
@piux2 what do you think about validating the package that way here?

Reply:
I think I will simply revert my last commit to only check for emptiness. Introducing `Validate` here would require too many changes and is a bit out of scope.

Reply:
We do not need to check isEmpty at all for the package. 

https://github.com/gnolang/gno/blob/61520301d58a6ce0ace86fefefab2f29e66b7d06/gnovm/pkg/gnolang/nodes.go#L1112

this will not read any empty directory. No package node and pkg name will be created if the directory is empty.

![image](https://github.com/gnolang/gno/assets/90544084/0a429eec-433b-4584-88fd-c16d05fc340b)

the valid package node with names are created as following:

bytealg
sha256
math
os

the following are not created as package nodes at the first place before changes in this commit.  

internal
crypto


Try to inject native method at internal, or crypto should not be allowed, and should panic in stdlibs/stdlib.go ( for on chain ) and tests/import.go ( for off chain testing). The native method can only be injected in a valid PackageNode.